### ISSUE TITLE
Support for reusable workflow

### ISSUE THREAD

**Issue by kvashchuka** [gitboard-io/gitboard-action#5]
Hi! Is it possible to use gitboard-io if I have reusable workflows? 

I am in a situation when I have a main workflow that calls another workflow like this:
```
jobs:
  run_reusable_wf:
    uses: ./.github/workflows/reusable_wf.yml
```
I tried to add gitboard-io call inside said reusable workflow, but I get following error in GA logs:
```
Run gitboard-io/gitboard-action@main
Gitboard.io Action configuration not provided, check username and api key - Unauthorized
```

And in the same repo it works well in the other workflow, which does not call a reusable workflow.

Is this functionality not supported or am I doing it wrong?

**Comment by GDownes** [comment]
H @kvashchuka,
Thanks for this,
Unfortunately this is a use-case we haven't tested.

Based on this issue we've:
 - Updated development roadmap to include support for reusable workflows.
 
 I'll keep you updated on progress

**Comment by GDownes** [comment]
Hi @kvashchuka 

We've now tested reusable workflows. 

Reusable-workflows are supported, but with limited use.

The error you've included highlights that the gitboard-action didn't get the required key property. In our testing we included `secrets: inherit` to ensure the reusable-workflow.yml job has access to our gitboard token contained within a secret, like below: 

```
  reusable:
    uses: ./.github/workflows/reusable-workflow.yml
    secrets: inherit
```

Any action is constrainted to the context of the calling  job. In the case of a reusable-workflow calling an action, the context of the action is the reusable-workflow.

The gitboard-action would only track the reusable workflow jobs success, failure & timings and not the containing parent jobs success, failure & timings. 

Putting the action inside a reusable workflow is unsuitable for its desired propose of tracking the parent jobs status. The gitboard-action needs to be ran from the job you want to track. As a reusable-workflow spawn its own job their usefulness is limited for tracking build progress using an action.

Thanks for your support
